src/messages/FileMessage.vue

```diff
@@ -1,21 +1,60 @@
 <template>
-  <div class="sc-message--file" :style="messageColors">
-    <div class="sc-message--file-icon">
-      <img :src="data.file.url" class="sc-image" />
+  <div class="sc-message--file" :style="messageColors" :id="data.file.id">
+    <div class="sc-message--file-icon" v-if="data.file.mime == 'application/vnd.openxmlformats-officedocument.wordprocessingml.document'">
+      <ApplicationWord />
     </div>
-    <div class="sc-message--file-name" :style="messageColors">
-      <a :href="data.file.url ? data.file.url : '#'" target="_blank">{{ data.file.name || '' }}</a>
+    <div class="sc-message--file-icon" v-else-if="data.file.mime == 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'">
+      <ApplicationExcel />
     </div>
```

Comment:
Here and below: please use strict-equality operator (`===`) instead of `==`.

I think this whole block can also be replaced with a simple dynamic component tag:

```vue
<div class="sc-message--file-icon">
  <component :is="`Application${someMethodThatMapsMimeToIconSuffix()}`" />
</div>
```